unit_test/hip/Test_HIP.hpp

```diff
@@ -0,0 +1,21 @@
+#include <gtest/gtest.h>
+#include <Kokkos_Core.hpp>
+#include <KokkosKernels_config.h>
+
+#if defined(KOKKOSKERNELS_TEST_ETI_ONLY) && !defined(KOKKOSKERNELS_ETI_ONLY)
+#define KOKKOSKERNELS_ETI_ONLY
+#endif
+
+class hip : public ::testing::Test {
+protected:
+  static void SetUpTestCase()
+  {
+  }
+
+  static void TearDownTestCase()
+  {
+  }
+};
+
+#define TestCategory hip
+#define TestExecSpace Kokkos::Experimental::HIPSpace
```

Comment:
I think this should be HIP, not HIPSpace. You can't do e.g. ``HIPSpace().fence()``, and a lot of our tests do ``DEVICE().fence()``. This could also be fixed by canonicalizing to an execution space with ``DEVICE::execution_space().fence()``, but I think in general we don't support plugging in a memory space as device.

Reply:
Yes that's clearly a bad oversight and this might actually allow me to revert some changes I made to the team unit-tests